packages/docprovider/src/yprovider.ts

```diff
@@ -41,17 +44,28 @@ export class WebSocketProviderWithLocks
     this._path = options.path;
     this._contentType = options.contentType;
     this._serverUrl = options.url;
-    const color = '#' + env.getParam('--usercolor', getRandomColor().slice(1));
-    const name = env.getParam('--username', getAnonymousUserName());
+
     const awareness = options.ymodel.awareness;
     const currState = awareness.getLocalState();
-    // only set if this was not already set by another plugin
-    if (currState && currState.name == null) {
-      options.ymodel.awareness.setLocalStateField('user', {
-        name,
-        color
-      });
-    }
+    let color = '#' + env.getParam('--usercolor', getRandomColor().slice(1));
+    let name = env.getParam('--username', getAnonymousUserName());
```

Comment:
It appears to me (but I may be wrong) that the current implementation prevents the user from changing their username and color once it was saved in the StateDB. However, if user made a typo in their username I imagine they would like to modify it and override the changes present in the StateDB. This would require replacing the default value for `color` and `name` with e.g. `null`, moving the randomly-generated value to `else` statement below, and only using the value from StateDB if the user has not provided values via environment params. Does it make sense?

Reply:
This makes sense. Some changes are made in https://github.com/jupyterlab/jupyterlab/pull/10490/commits/bd9ded55c4fc5b613dc6ddb7a9a4875e654884ef, let me know if any thing still looks odd there. Thank you